src/snowflake/snowpark/mock/analyzer.py

```diff
@@ -751,24 +751,18 @@ def do_resolve_with_resolved_children(
             )
 
         if isinstance(logical_plan, TableUpdate):
-            raise NotImplementedError(
-                "[Local Testing] Table update is not implemented."
-            )
+            return MockExecutionPlan(logical_plan, self.session)
```

Comment:
Nit: these `if`s can be combined to a single `if` now.

Reply:
Good point, we will address these in a separate PR before merging back.